pkg/controllers/nodeclass/validation.go

```diff
@@ -75,38 +109,81 @@ func (n Validation) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (
 	}
 	tags, err := utils.GetTags(nodeClass, nodeClaim, options.FromContext(ctx).ClusterName)
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("getting tags, %w", err)
+		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeValidationSucceeded, "TagValidationFailed", err.Error())
+		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("validating tags, %w", err))
 	}
 
+	type validationFunc func(context.Context, *v1.EC2NodeClass, *karpv1.NodeClaim, map[string]string) (bool, error)
+	for conditionReason, f := range map[string]validationFunc{
+		ConditionReasonCreateFleetAuthFailed:          v.validateCreateFleetAuthorization,
+		ConditionReasonCreateLaunchTemplateAuthFailed: v.validateCreateLaunchTemplateAuthorization,
+		ConditionReasonRunInstancesAuthFailed:         v.validateRunInstancesAuthorization,
+	} {
+		if ok, err := f(ctx, nodeClass, nodeClaim, tags); err != nil {
+			return reconcile.Result{}, err
+		} else if !ok {
+			v.cache.SetDefault(v.cacheKey(nodeClass), conditionReason)
```

Comment:
Does the reason matter in the cache? Why not have a single, generic validation message that presents the variety of failures that can happen like we do on `validation.go:77`?